.github/workflows/main.yml

```diff
@@ -1,16 +1,39 @@
 name: Bump version
+
 on:
   push:
     branches:
       - master
+
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     steps:
-    - uses: actions/checkout@master
-      with:
-        fetch-depth: '0'
-    - name: Bump version and push tag
-      uses: anothrNick/github-tag-action@master
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+
+      - uses: stefanzweifel/git-auto-commit-action@v4
+        id: git_auto_commit
+        with:
+          commit_message: Bump version
+
+      - name: version-tag
+        uses: anothrNick/github-tag-action@1.40.0 # another pr require to move this to v1
+        if: steps.git_auto_commit.outputs.changes_detected == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: version-tag-major
```

Comment:
I'm not sure I understand what this is doing or what value it has.

- @v1 will always use the latest v1.x.x tag.
- The below script should have a comment describing what it's doing and why we can't use this very action to achieve it.

Reply:
Here https://github.com/anothrNick/github-tag-action/tags everytime we create a new release we sould also update the `major` tag to reflec the change of the latest minor/patch.

So when consumer want to refer to @1 can do it in the same way as @1.42.0

if you use tools like https://github.com/sethvargo/ratchet then using @1 is convinient same as with checkout@v3 and not checkout@v3.0.2

Reply:
Is a different purpose with this very action this action bumps major, does not updates existing tags to update the reference

Reply:
![image](https://user-images.githubusercontent.com/23391642/189569969-0aa09883-0b5f-4430-8e07-b023e9e13661.png)
checkout example

Reply:
I did the same here https://github.com/bridgecrewio/checkov-action/pull/92

Reply:
Ohhhh I've got you! Understood.